ServerCore/Pages/Submissions/FreeformQueue.cshtml.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Submissions
+{
+    [Authorize(Policy = "IsEventAdminOrEventAuthor")]
+    public class FreeformQueueModel : EventSpecificPageModel
+    {
+        public FreeformQueueModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager)
+            : base(serverContext, userManager)
+        {
+        }
+
+        public Puzzle Puzzle { get; set; }
+
+        public class SubmissionView
+        {
+            public int PuzzleId { get; set; }
+            public string PuzzleName { get; set; }
+            public string TeamName { get; set; }
+            public string SubmissionText { get; set; }
+            public int SubmissionId { get; set; }
+        }
+
+        [BindProperty]
+        public string FreeformResponse { get; set; }
+
+        [BindProperty]
+        public int SubmissionID { get; set; }
+
+        public enum FreeformResult
+        {
+            None,
+            Accepted,
+            Rejected
+        }
+
+        [BindProperty]
+        public FreeformResult Result { get; set; }
+
+        public List<SubmissionView> Submissions { get; set; }
+
+        public async Task OnGetAsync(int? puzzleId)
+        {
+            await PopulateUI(puzzleId);
+        }
+
+        private async Task PopulateUI(int? puzzleId)
+        {
+            if (puzzleId != null)
+            {
+                Puzzle = await _context.Puzzles.Where(m => m.ID == puzzleId).FirstOrDefaultAsync();
+            }
+
+            // todo: think about a way to go back to old submissions and redo them (e.g. to fix a forgotten comment)
```

Comment:
This also needs to have an audit trail, by which I only mean a way to check up on a grader and confirm that they are grading as you would expect. Also selection of greatest hits like Safari does; consider adding a "Favorite" checkbox that can be filtered later. Not a blocker for tonight if it slows things down.

Reply:
Great ideas. I'll start thinking about that for a followup. Filed #608